src/message/message-collection.ts

```diff
@@ -75,3 +81,14 @@ export async function getMessages(channel: ChannelID, take: number = 10, after?:
         next: docs[docs.length - 1],
     };
 }
+
+export async function updateMessage(channelId: ChannelID, messageId: MessageID, sender: UserID, data: IMessageData): Promise<IMessage> {
+    const message: IMessageRecord = {
+        message: data.message,
+        payload: JSON.stringify(data.payload || null),
+        sender: sender,
+        createdAt: Date.now(),
+    };
+    await setDoc(_messageRef(channelId, messageId), message);
```

Comment:
1. Is it OK to use `setDoc`? It will completely rewrite the whole message so you should gather all the information to update
2. You overwrite the `createdAt` fields which ruins the idea. I think it should remain the same. Adding `updatedAt` filed could be a better idea